firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java

```diff
@@ -211,29 +212,29 @@ boolean getIsDeviceAllowedToSendNetworkEvents() {
   static class RateLimiterImpl {
 
     private static final AndroidLogger logger = AndroidLogger.getInstance();
-    private static final long MICROS_IN_A_SECOND = TimeUnit.SECONDS.toMicros(1);
+    private static final long MICROS_IN_A_SECOND = SECONDS.toMicros(1);
 
     private final Clock clock;
     private final boolean isLogcatEnabled;
 
     // Last time a token is consumed.
-    private Timer lastTimeTokenConsumed;
+    private Timer lastTimeTokenReplenished;
 
-    // Number of new tokens generated per second.
-    private double rate;
+    // The Rate object representing the number of tokens generated per specified time unit
+    private Rate rate;
     // Token bucket capacity, also the initial number of tokens in the bucket.
     private long capacity;
     // Number of tokens in the bucket.
     private long tokenCount;
 
-    private double foregroundRate;
-    private double backgroundRate;
+    private Rate foregroundRate;
+    private Rate backgroundRate;
 
     private long foregroundCapacity;
     private long backgroundCapacity;
 
     RateLimiterImpl(
-        final double rate,
+        final Rate rate,
```

Comment:
I'm noticing further down that there is a separate foreground and background rate, but they seem to be ignoring this initial one and just blindly overwriting it. Do you understand how it works?

Reply:
I think the foreground and background rate is set during initilization and read from remote config, but the passed in rate is still in use, until changeRate() is called. changeRate() is called by the TransportManager when an app updates its state (i.e., changes from foreground to background). At least that's my understanding.

Reply:
But shouldn't the passed-in rate be either foreground or background? Why would there be a 3rd rate that is only used to initialize and then overwritten?

Reply:
Perhaps because we don't know whether the app will be in foreground or background when it finishes initializing? I am not sure if there were other reasons that I am not aware of. We could remove the passed in rate and set it to either foreground or background on initialization if that makes more sense (but which one?)

Reply:
@visumickey do you know why this is the way it is and whether it makes sense to keep it or change it?

Reply:
That sounds good - I'd recommend leaving it out of this PR though to not continue changing more things in a single PR.

Reply:
Sounds good, I will leave that out of this PR. Although, is there a possibility that the app has not finished initializing yet so it's not in either foreground or background?

Reply:
Yes, that is a possibility. But in those situations, it is fine to default it to the foreground rate.